docs/sources/flow/reference/config-blocks/http.md

```diff
@@ -0,0 +1,143 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/config-blocks/http/
+title: http
+---
+
+# http block
+
+`http` is an optional configuration block used to customize how Grafana Agent's
+HTTP server functions. `http` is specified without a label and can only be
+provided once per configuration file.
+
+{{% admonition type="note" %}}
+While the `http` block can reference component exports, some components that
+rely on the HTTP server have a hidden dependency on the `http` block that may
+result in a circular dependency error.
+
+Only references to components named `remote.*` or `local.*` are guaranteed to
+work without any circular dependency errors.
+{{% /admonition %}}
+
+## Example
+
+```river
+http {
+  tls {
+    cert_file = env("TLS_CERT_FILE_PATH")
+    key_file  = env("TLS_KEY_FILE_PATH")
+  }
+}
+```
+
+## Arguments
+
+The `http` block supports no arguments and is configured completely through
+inner blocks.
+
+## Blocks
+
+The following blocks are supported inside the definition of `http`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+tls | [tls][] | Define TLS settings for the HTTP server. | no
+
+[tls]: #tls-block
+
+### tls block
+
+The `tls` block configures TLS settings for the HTTP server.
+
+{{% admonition type="warning" %}}
+If the `tls` block is added and the configuration is reloaded when Grafana
+Agent is running, existing connections will continue to communicate over
+plaintext. Similarly, if the `tls` block is removed and the configuration is
+reloaded when Grafana Agent is running, existing connections will continue to
+communicate over TLS.
+
+To ensure all connections use TLS, start Grafana Agent with the `tls` block
+already configured.
+{{% /admonition %}}
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`cert_pem` | `string` | PEM data of the server TLS certificate. | `""` | conditionally
+`cert_file` | `string` | Path to the server TLS certificate on disk. | `""` | conditionally
+`key_pem` | `string` | PEM data of the server TLS key. | `""` | conditionally
+`key_file` | `string` | Path to the server TLS key on disk. | `""` | conditionally
```

Comment:
Is the reason we are enabling file is so that it can check it on each access like it does today? Normally I would say use local.file or remote.file and only use expose *_pem.

Reply:
That, plus parity and alignment in general: all of our other TLS-related settings support files plus strings.